pkg/mempool/simplemempool/internal/parts/computeids/computeids.go

```diff
@@ -19,13 +19,27 @@ import (
 func IncludeComputationOfTransactionAndBatchIDs(
 	m dsl.Module,
 	mc common.ModuleConfig,
-	_ *common.ModuleParams,
+	params *common.ModuleParams,
 	_ *common.State,
 ) {
 	mppbdsl.UponRequestTransactionIDs(m, func(txs []*trantorpbtypes.Transaction, origin *mppbtypes.RequestTransactionIDsOrigin) error {
-		txMsgs := make([]*hasherpbtypes.HashData, len(txs))
-		for i, tx := range txs {
-			txMsgs[i] = &hasherpbtypes.HashData{Data: serializeTXForHash(tx.Pb())}
+		if txs == nil || len(txs) > params.MaxTransactionsInBatch {
+			// Invalid request, ignore
```

Comment:
This is not an external message (or could it be based on one?), but an internal event. If this check fails, that means that something is wrong with the implementation and we should return an error here.

Reply:
This is called as part of an external message received [UponProvideBatchMessageReceived](https://github.com/filecoin-project/mir/blob/8f352300a01bb71d1bc05cb62a92e0e105ad0862/pkg/availability/multisigcollector/internal/parts/batchreconstruction/batchreconstruction.go#L128-L142)

Reply:
I see. But in that case, if we really do want to enforce this predicate on the received batch, I would prefer enforcing it at the reception of the message.

In principle, from the point of view of the ID computation code, IDs can be computed for both empty batches and large batches, so I find it much cleaner to not enforce this condition here. This part of the code does not seem to rely on it, so it should not enforce it.

Reply:
MaxTransactionsInBatch is a parameter of the mempool that the availability module does not even know about (and should not have to imo). I think it is conceptually sensible to make the check in the mempool and not in the availability module. We can remove the check with nil, but at the moment it is also a condition that no nil transaction is given as a request. I am happy to remove the `== nil`. 

I have added a log in the event of that check returning nil.

Reply:
That's fair. In that case, let's keep this and the `nil` check here. I'd still also print a warning though when ignoring the whole request because of a `nil` transaction (also suggested separately).

Also, feel free to resolve this one after reading.

Reply:
In fact, this condition is wrong. I guess you meant `||` instead of `&&`.
But anyway, I still think that it's better to remove the `len(txs) == 0` check. I see no strong reason for the mempool to not work for empty batches.
What I meant by the `nil` check is checking that no transactino in the list is `nil` (the code just below). This, as you convinced me, we should keep.